build.gradle

```diff
@@ -735,3 +736,88 @@ task distRpm(dependsOn: "jpackageImage", type: Rpm) {
 	prefix '/opt' // Relocations: /opt
 	arch = 'x86_64'
 }
+
+task dockerPullOracleLinux8(type: DockerPullImage) {
+	group = "openjdk17-oraclelinux8 container"
+	image = "openjdk:17-oraclelinux8"
+	description = "Pulls a docker image for oraclelinux8 from ${image}."
+}
+
+task dockerCreateOracleLinux8(dependsOn: dockerPullOracleLinux8, type: DockerCreateContainer) {
+	group = "openjdk17-oraclelinux8 container"
+	description = "Creates an OracleLinux8 container named ${oracleLinux8ContainerName}."
+	imageId = dockerPullOracleLinux8.getImage()
+	containerName = oracleLinux8ContainerName
+	hostConfig.binds = ["${projectDir}":"/source/anet"]
+	stdinOpen = true
+}
+
+task dockerStartOracleLinux8(type: DockerStartContainer) {
+	group = "openjdk17-oraclelinux8 container"
+	description = "Starts ${oracleLinux8ContainerName} container."
+	containerId = oracleLinux8ContainerName
+}
+
+task dockerStopOracleLinux8(type: DockerStopContainer) {
+	group = "openjdk17-oraclelinux8 container"
+	description = "Stops ${oracleLinux8ContainerName} container."
+	containerId = oracleLinux8ContainerName
+}
+
+task dockerRemoveOracleLinux8(type: DockerRemoveContainer) {
+	group = "openjdk17-oraclelinux8 container"
+	description = "Removes ${oracleLinux8ContainerName} container."
+	containerId = oracleLinux8ContainerName
+}
+
+task dockerInstallYumOracleLinux8(type: DockerExecContainer) {
+	group = "openjdk17-oraclelinux8 container"
+	description = "Install git yum to ${oracleLinux8ContainerName} container."
+	containerId = oracleLinux8ContainerName
+	commands = [["sh", "-c", "microdnf -y install git yum "] as String[]]
+}
+
+task dockerInstallDevToolsOracleLinux8(dependsOn: dockerInstallYumOracleLinux8, type: DockerExecContainer) {
+	group = "openjdk17-oraclelinux8 container"
+	description = "Install Development Tools to ${oracleLinux8ContainerName} container."
+	containerId = oracleLinux8ContainerName
+	commands = [ ["sh", "-c", "yum -y groupinstall \"Development Tools\""] as String[] ]
+}
+
+task dockerClearCachesInLinux8( dependsOn: dockerInstallDevToolsOracleLinux8, type: DockerExecContainer) {
+	group = "openjdk17-oraclelinux8 container"
+	description = "Clear caches in ${oracleLinux8ContainerName} container."
+	containerId = oracleLinux8ContainerName
+	user = "root"
```

Comment:
Should this run as `root`? The original script made sure that all files that were created were owned by the user executing the script. Now I will end up with files in my development directory owned by `root` after running these steps. That shouldn't be the case.

Reply:
Actually it is not an obligation. It was done to remove gradle caches under root folder (rm -rf /root/.gradle/caches). if we skip this part there is no need the root user.